File: app/src/main/java/org/wikipedia/views/SearchEmptyView.java
package org.wikipedia.views;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.annotation.StringRes;

import org.wikipedia.databinding.ViewSearchEmptyBinding;

public class SearchEmptyView extends LinearLayout {
    private TextView emptyText;

    public SearchEmptyView(Context context) {
        super(context);
        init();
    }

    public SearchEmptyView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public SearchEmptyView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    public SearchEmptyView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init();
    }

    public void setEmptyText(@StringRes int id) {
        emptyText.setText(id);
    }

    public void setEmptyText(@Nullable CharSequence text) {
        emptyText.setText(text);
    }

    private void init() {
        ViewSearchEmptyBinding binding = ViewSearchEmptyBinding.bind(this);
        emptyText = binding.searchEmptyText;
        setOrientation(VERTICAL);
    }
}

File: app/src/main/java/org/wikipedia/views/SearchEmptyView.kt
package org.wikipedia.views

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.LinearLayout
import androidx.annotation.StringRes
import org.wikipedia.databinding.ViewSearchEmptyBinding

class SearchEmptyView(context: Context, attributeSet: AttributeSet) : LinearLayout(context, attributeSet) {

    private val binding = ViewSearchEmptyBinding.inflate(LayoutInflater.from(context), this)

    init {
        orientation = VERTICAL
    }

    fun setEmptyText(@StringRes id: Int) {
        binding.searchEmptyText.setText(id)
    }

    fun setEmptyText(text: CharSequence?) {
        binding.searchEmptyText.text = text
    }
}
